server/src/main/java/io/crate/metadata/view/ViewInfo.java

```diff
@@ -45,14 +46,21 @@ public class ViewInfo implements RelationInfo {
     private final RelationName ident;
     private final String definition;
     private final List<Reference> columns;
+    private final List<Reference> subColumns;
     private final String owner;
     private final SearchPath searchPath;
 
     @VisibleForTesting
-    public ViewInfo(RelationName ident, String definition, List<Reference> columns, @Nullable String owner, SearchPath searchPath) {
+    public ViewInfo(RelationName ident,
+                    String definition,
+                    List<Reference> columns,
+                    List<Reference> subColumns,
```

Comment:
I'd prefer adding sub columns already to the given `columns` list (maybe also rename it to `references`?) and filter subcolumns out at the `columns()` function. I think this would simplify all a bit, resolve any confusion why top-level and sub-columns are treated different and avoid adding another argument here.

Reply:
Done in https://github.com/crate/crate/pull/16165/commits/842e0654ddf0aa3705e5c04f7cf9288186a3644b.

Not sure about simplicity though: now each `columns()` call creates a new list.